apps/condo/domains/user/integration/telegram/routes.js

```diff
@@ -0,0 +1,75 @@
+const crypto = require('crypto')
+
+const { v4: uuid } = require('uuid')
+
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getRedisClient } = require('@open-condo/keystone/redis')
+
+const TelegramAuthBotController = require('@condo/domains/user/integration/telegram/BotController')
+const {
+    TELEGRAM_AUTH_STATUS_PENDING,
+    TELEGRAM_AUTH_STATUS_ERROR,
+    TELEGRAM_AUTH_STATUS_SUCCESS,
+    TELEGRAM_AUTH_REDIS_START,
+    TELEGRAM_AUTH_REDIS_TOKEN,
+} = require('@condo/domains/user/integration/telegram/constants')
+const { getUserType } = require('@condo/domains/user/integration/telegram/utils')
+
+const TELEGRAM_AUTH_BOT_URL = process.env.TELEGRAM_AUTH_BOT_URL
+
+const logger = getLogger('telegram-auth')
+const redisClient = getRedisClient()
+
+class TelegramAuthRoutes {
+    #telegramAuthBot = new TelegramAuthBotController()
+
+    constructor () {
+        this.#telegramAuthBot.init()
+    }
+
+    async startAuth (req, res, next) {
+        try {
+            if (!TELEGRAM_AUTH_BOT_URL) throw new Error('TELEGRAM_AUTH_BOT_URL is not configured')
+            const userType = getUserType(req)
+            const startKey = uuid()
+            const uniqueKey = crypto.randomBytes(32).toString('hex')
+            const startLink = `${TELEGRAM_AUTH_BOT_URL}?start=${startKey}`
+
+            await Promise.all([
+                redisClient.set(`${TELEGRAM_AUTH_REDIS_START}${startKey}`, JSON.stringify({ uniqueKey, userType }), 'EX', 300),
```

Comment:
Maybe make a small utility for generating a key, so as not to generate it manually each time?